api/src/main/java/org/pmiops/workbench/api/WorkspacesController.java

```diff
@@ -346,11 +346,24 @@ private boolean checkWorkspaceWriteAccess(String workspaceNamespace, String work
   }
 
   @Override
-  public ResponseEntity<Void> deleteWorkspace(String workspaceNamespace, String workspaceId) {
+  public ResponseEntity<EmptyResponse> deleteWorkspace(String workspaceNamespace, String workspaceId) {
     org.pmiops.workbench.db.model.Workspace dbWorkspace = workspaceService.getRequired(
         workspaceNamespace, workspaceId);
+    try {
+      fireCloudService.deleteWorkspace(workspaceNamespace, workspaceId);
+    } catch (org.pmiops.workbench.firecloud.ApiException e) {
+      if (e.getCode() == 403) {
+        throw new ForbiddenException(String.format("Insufficient permissions to delete workspace %s/%s",
+            workspaceNamespace, workspaceId));
+      } else if (e.getCode() == 404) {
```

Comment:
I'm not sure you should throw in this case. It's possible that somebody deletes from FC but not AofU (either in their UI or if this code dies right after calling FC.)

The tricky part is... what happens if you don't have read permission on the workspace and try to delete it anyway? Does FC return a 404, or a 403?

If it returns a 404, we may want to check our representation of the ACL, and continue with the delete from our DB if the user is an owner of the workspace.

Reply:
FC returns a 404 if you don't have access at all to a workspace. Would you be up for doing a sync up call on this later today? I think there may still be an open question on how we want to handle this. (If we get a 404 from FC, and have a representation in our ACL, there are two options for the situation: 1) it got deleted in FC, which means old data from our DB is still hanging around, or 2) the user was unshared from the workspace, but that ACL change hasn't propegated up to our ACL representation yet. I personally think 2) is more dangerous, and we should check for 1) when we do an ACL update?)

Reply:
This is only a problem if the user is updating sharing and deleting workspaces in FC, and not in our UI, I believe?

Reply:
As discussed offline: we have a general issue around orphaned AofU workspaces that will probably require some batch job. 

This logic should be fine.